volatility3/framework/plugins/windows/psscan.py

```diff
@@ -261,40 +261,62 @@ def _generator(self):
+                    )[0]
                 else:
-                    try:
-                        vproc = self.virtual_process_from_physical(
-                            self.context,
-                            kernel.layer_name,
-                            kernel.symbol_table_name,
-                            proc,
-                        )
-                    except exceptions.PagedInvalidAddressException:
-                        vproc = None
+                    # the display should be virtual addresses, so proc can be used
+                    offset = proc.vol.offset
 
-                file_output = "Error outputting file"
-                if vproc:
-                    file_handle = pslist.PsList.process_dump(
-                        self.context,
-                        kernel.symbol_table_name,
-                        pe_table_name,
-                        vproc,
-                        self.open,
-                    )
+                #  renderers.UnreadableValue()
+            else:
+                # proc is in virtual mem, so a new object needs to be creatd.
+                vproc = self.virtual_process_from_physical(
+                    self.context, kernel.layer_name, kernel.symbol_table_name, proc
+                )
+                if self.config["physical"]:
+                    # the display should be physical addresses, so proc can be used
+                    # as it is
+                    offset = proc.vol.offset
+                else:
+                    # the display should be virtual address, so vproc should be used
+                    # however virtual_process_from_physical is not always able to create
+                    # a vproc, in that case we need to display a UnreadableValue()
+                    if vproc is not None:
+                        offset = vproc.vol.offset
+                    else:
+                        offset = renderers.UnreadableValue()
 
-                    if file_handle:
-                        file_output = file_handle.preferred_filename
+            if self.config["dump"]:
+                file_handle = pslist.PsList.process_dump(
+                    self.context,
+                    kernel.symbol_table_name,
+                    pe_table_name,
+                    vproc,
+                    self.open,
+                )
+                file_output = "Error outputting file"
+                if file_handle:
+                    file_output = file_handle.preferred_filename
 
-            if not self.config["physical"]:
-                offset = proc.vol.offset
-            else:
-                (_, _, offset, _, _) = list(
-                    memory.mapping(offset=proc.vol.offset, length=0)
-                )[0]
+            # format offset for display, but catch errors when UnreadableValue
+            # cannot be formatted
+            try:
+                display_offset = format_hints.Hex(offset)
```

Comment:
We could set the offset to None rather than `UnreadableValue` and then the test will be simpler/more efficient than setting up an exception handler to catch a known/expected exception?

Reply:
Yup - that's a much better way to do it.